src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -120,23 +121,17 @@
      * counts the number of assessments of a course, which are either rated or not rated
      *
      * @param exerciseIds - the exercises of the course
-     * @param rated     - only counts assessments which are either rated or not rated
-     * @return count of rated/unrated assessments of a course
+     * @return an array with 2 elements: count of rated (in time) and unrated (late) assessments of a course
```

Comment:
I suppose this comment is no longer correct, since we might return a list with three entries? (One for `null`)

Reply:
> I find returning a list with three "hardcoded" entries a bit odd, but don't know a better way to do this.
> 
> Tested on TS1 and it works fine. It also works with my local data from last semester, but PGdP still takes 25.3s and 43.4s for both dashboard calls. (Which is still an improvement to before I suppose.)

I also don't really like the solution but did not find a better one to combine the two calls into one.
I guess further improvements are possible and we should further try to find ways how to speedup the whole process.
The good thing is that the PGdP assessment dashboard on production is loading faster than on the local environment:
* `for-assessment-dashboard` takes around 2.3s (before the changes in this PR)
* `stats-for-assessment-dashboard` takes around 10.9s (before the changes in this PR)

I guess after merging this PR those values will be lower, `stats-for-assessment-dashboard` should then take less than 6.5s if the performance gain is comparable